server/go/server.go

```diff
@@ -97,6 +98,33 @@ func handlePublicKey(w http.ResponseWriter, r *http.Request) {
 	})
 }
 
+func handleGetCheckoutSession(w http.ResponseWriter, r *http.Request) {
+	if r.Method != "GET" {
+		http.Error(w, http.StatusText(http.StatusMethodNotAllowed), http.StatusMethodNotAllowed)
+		return
+	}
+
+	keys, ok := r.URL.Query()["sessionId"]
+
+    if !ok || len(keys[0]) < 1 {
+        log.Println("CheckoutSession ID is missing from URL", r.URL.Query())
```

Comment:
Put `http.Error(w, http.StatusText(http.StatusBadRequest), http.StatusBadRequest)` as well? It will return a 200 status code in its current state.